From 1b818a33e3cd20250b72306946789d09d2c7b2c8 Mon Sep 17 00:00:00 2001
From: Hal Feng <hal.feng@starfivetech.com>
Date: Sun, 26 Jan 2025 16:56:05 +0800
Subject: [PATCH 25/55] spi: pl022: Fix spi overlay falut

set_cs() should not be set for spi controller when using overlay.
clk_disable_unprepare() is not needed after using
devm_clk_get_enabled() to get and enable clocks.

Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
---
 drivers/spi/spi-pl022.c | 3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/spi/spi-pl022.c
+++ b/drivers/spi/spi-pl022.c
@@ -1908,7 +1908,6 @@ static int pl022_platform_probe(struct p
 	/* If open CONFIG_PM, auto_runtime_pm should be false when of-platform.*/
 	host->auto_runtime_pm = true;
 	host->transfer_one = pl022_transfer_one;
-	host->set_cs = pl022_cs_control;
 	host->handle_err = pl022_handle_err;
 	host->unprepare_transfer_hardware = pl022_unprepare_transfer_hardware;
 	host->rt = platform_info->rt;
@@ -2435,8 +2434,6 @@ static void starfive_of_pl022_remove(str
 	if (pl022->host_info->enable_dma)
 		pl022_dma_remove(pl022);
 
-	clk_disable_unprepare(pl022->clk);
-
 	pm_runtime_put_noidle(&pdev->dev);
 	pm_runtime_disable(&pdev->dev);
 	pm_runtime_set_suspended(&pdev->dev);
